src/main/java/ch/hftm/controllers/AlbumController.java

```diff
@@ -27,8 +45,76 @@ private void setupAlbumPage() {
         if (album == null) {
             return;
         }
-        albumTitle.setText(album.getName());
-        // TODO: Get all images of album and display them nicely in a grid
+        albumTitle.setText("Album: " + album.getName());
+        initializeAlbum();
+    }
+
+    /**
+     * Create a pane to display an image including the meta attribut descriptors and values
+     */
+    private Pane createImagePane(Image image, Pane samplePane) {
+        Pane imagePane = new Pane();
+        imagePane.setPrefHeight(samplePane.getPrefHeight());
+        imagePane.setPrefWidth(samplePane.getPrefWidth());
+
+        ImageView sampleImageView = (ImageView) samplePane.lookup("#imageView");
+        ImageView imageView = new ImageView();
+        imageView.setFitHeight(sampleImageView.getFitHeight());
+        imageView.setFitWidth(sampleImageView.getFitWidth());
+        javafx.scene.image.Image fxImage = new javafx.scene.image.Image(new File(image.getFullPath()).toURI().toString());
+        imageView.setImage(fxImage);
+
+        GridPane imageMetaPane = new GridPane();
+        ColumnConstraints col1 = new ColumnConstraints();
+        col1.setPercentWidth(35.0);
+        ColumnConstraints col2 = new ColumnConstraints();
+        col2.setPercentWidth(65.0);
+        imageMetaPane.getColumnConstraints().add(col1);
+        imageMetaPane.getColumnConstraints().add(col1);
+
+        // Add an amount of descriptor labels to the grid based on the amount of the most important meta values
+        for (int i = 0; i < ATTR_AMOUNT; i++) {
+            Label imageAttr = new Label();
+            imageAttr.setText(String.format("Attribute %d:", i));
+            imageAttr.setVisible(true);
+            imageMetaPane.add(imageAttr, 0, i);
+        }
+
+        // Add an amount of value labels to the grid based on the amount of the most important meta value
+        for (int i = 0; i < ATTR_AMOUNT; i++) {
```

Comment:
Is there a reason that we have to loop twice? Could we not create both labels in the same loop? If you want to separate the code you can create a create a "createLabel" private method which you can then call twice in the same loop, once for each label type